api/controllers/ProjectController.js

```diff
@@ -1,188 +1,230 @@
-var crypto = require('crypto'),
-  slug = require('slug');
+var slug = require('slug')
 
-var makeSlug = function(title) {
-  if (!title || title == 'New Project') return 'untitled';
-  return slug(title, {mode: 'rfc3986'});
-};
+var makeSlug = function (title) {
+  if (!title || title === 'New Project') return 'untitled'
+  return slug(title, {mode: 'rfc3986'})
+}
 
-var editableAttributes = [
-  'community_id', 'title', 'intention', 'details', 'video_url', 'image_url', 'visibility', 'location'
-];
+var editableAttributeNames = [
+  'community_id', 'title', 'intention', 'details', 'visibility', 'location'
+]
+
+var mediaAttributeNames = [
+  'image_url', 'thumbnail_url', 'video_url'
+]
 
 var projectRelations = [
   {user: qb => qb.column('id', 'name', 'avatar_url')},
   {community: qb => qb.column('id', 'name', 'avatar_url')},
   {contributors: qb => qb.column('users.id')},
   {posts: qb => {
-    qb.column('post.id');
-    qb.where('type', Post.Type.REQUEST);
-  }}
-];
-
-var projectAttributes = project => _.extend(project.toJSON(), {
-  contributor_count: project.relations.contributors.length,
-  open_request_count: project.relations.posts.length
-});
-
-var maybeGenerateVideoThumbnail = Promise.method(function(attrs) {
-  if (attrs.video_url) {
-    return Project.generateThumbnailUrl(attrs.video_url)
-    .then(url => attrs.thumbnail_url = url);
+    qb.column('post.id')
+    qb.where('type', Post.Type.REQUEST)
+  }},
+  'media'
+]
+
+var projectAttributes = function (project) {
+  var attrs = project.toJSON()
+  attrs = _.extend(attrs, {
+    contributor_count: project.relations.contributors.length,
+    open_request_count: project.relations.posts.length
+  })
+  attrs = _.extend(attrs, mediaAttributes(project))
+  return attrs
+}
+
+var mediaAttributes = function (project) {
+  var attrs = project.toJSON()
```

Comment:
better not to have project.toJSON involved here, because it's redundant when this is used in projectAttributes. and then in findOne, just continue to use project.toJSON and extend it with mediaAttributes